### PR TITLE
Rename and refactor config plugin as response-data plugin

### PR DESCRIPTION
**Pull request by agerard-godaddy**
## Summary

We have had a lot of confusion around the config plugin and it's purpose. As such, we are renaming and refocusing what the plugin does. That is, to allow environment specific data to be accessible for requests, with public data available with responses.

Instead of the generic 'config' name, we will term this gasketData which pairs well with the `@gasket/data` package - which is what makes this data accessible in browser code. We are dropping the `redux` property, aligning on `public` which will be added to the Redux state as `gasketData` for apps that choose to opt-in to Redux.

The lifecycles have been renamed also.
- `gasketData` can be used to tune up the base object during app `init`.
- `responseData` can be used to tune up the "public" data uniquely for each response.

## Changelog
**@gasket/plugin-response-data**
- Renamed from @gasket/plugin-config
- Renamed lifecycles `gasketData` and `responseData`
- Renamed expected setup config to `gasket-data.config.js`
- Renamed expected setup env file dir to `./gasket-data`
- Add `@gasket/data` during create lifecycle

## Test Plan

- [x] Updat unit tests
- [x] Test with create command
- [ ] Test with local app update

### CODE REVIEW

File: packages/gasket-plugin-response-data/lib/middleware.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        }
+      );
+
+      // Error handling
+      // Better error description for the responseData lifecycles
+      // In the event a config object isn't returned from the hook(s)
+      // it will result in an error in the destructuring of req.gasketData
+      if (!publicGasketData) {
+        throw new Error(
+          'An responseData lifecycle hook did not return a config object.'
```
Comment (mmason2-godaddy) [inline packages/gasket-plugin-response-data/lib/middleware.js:34]:
```suggestion
          'A responseData lifecycle hook did not return a config object.'
```

File: packages/gasket-plugin-response-data/lib/plugin.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    async preboot(gasket) {
+      const data = await gasket.execWaterfall(
+        'gasketData',
+        mergeRootConfig(gasket, mergeConfigFiles(gasket))
+      );
+
+      // Better error description for the lifecycles
+      if (!data) {
+        throw new Error(
+          'An gasketData lifecycle hook did not return a config object.'
```
Comment (mmason2-godaddy) [inline packages/gasket-plugin-response-data/lib/plugin.js:26]:
```suggestion
          'A gasketData lifecycle hook did not return a config object.'
```

**Review by mmason2-godaddy — APPROVED** [review]
🥇

**Review by jpina1-godaddy — APPROVED** [review]

File: packages/gasket-plugin-response-data/package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
```
[the remark this answers is not in the document]
Comment (agerard-godaddy) [reply packages/gasket-plugin-response-data/package.json:1]: Great catches! I fixed the hopeage link but the codeowners needs a complete overhaul. I'll handle that as a separate PR starting with the `main` branch.

[the remark this answers is not in the document]
Comment (agerard-godaddy) [reply packages/gasket-plugin-response-data/package.json:1]: Codeowners changes https://github.com/godaddy/gasket/pull/686

**Pull request merged**